polling_stations/apps/data_importers/management/commands/import_eoni_from_s3.py

```diff
@@ -0,0 +1,415 @@
+import csv
+import re
+from pathlib import Path
+
+import chardet
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+from pyproj import Transformer
+
+from data_importers.s3wrapper import parse_s3_uri, S3Wrapper
+from data_importers.management.commands.import_eoni import Command as EONI_Importer
+
+from polling_stations.settings.constants.importers import EONIImportScheme
+
+
+def attempt_decode(body: bytes):
+    sample = get_body_sample(body)
+    detection = chardet.detect(sample)
+    encoding = detection["encoding"]
+    if encoding == "utf-16le":
+        return body.decode(encoding), encoding
+    encodings = ["utf-8", "windows-1252", "latin-1"]
+    for encoding in encodings:
+        try:
+            return body.decode(encoding), encoding
+        except UnicodeDecodeError as e:
+            last_exception = e
+            continue
+    raise last_exception
+
+
+def get_body_sample(body: bytes):
+    if b"\r\n" in body:
+        line_sep = b"\r\n"
+    elif b"\n" in body:
+        line_sep = b"\n"
+    else:
+        return body
+    lines = body.split(line_sep)
+    if len(lines) > 20:
+        return line_sep.join(lines[:20])
+    return line_sep.join(lines)
+
+
+def is_eoni_export_key(key: str) -> bool:
+    """
+    Should return true for things like:
+    '2023-04-12-EONIextract.txt',
+    '2024-07-03-EONIextract-sample.txt',
+    '2024-11-21-EONIextract_NEW.txt'
+
+    """
+    eoni_export_key_pattern = re.compile(
+        r"""
+        ^           # Start of string
+        20[2-9]     # First 3 digits of year: 202-209
+        [0-9]       # Last digit of year: 0-9
```

Comment:
No. Was just in a fiddly mood when I did it. I generally use `[0-9]` rather than `\d` because it feels clearer to me. I obviously stopped myself from making the months `[0,1][0-9]`. I'm not bothered what we go for. As I said it's mostly a way of documenting what we expect the file names to be.